tests/PHPStan/Analyser/data/foreach-loop-no-scope-pollution.php

```diff
@@ -0,0 +1,115 @@
+<?php declare(strict_types=1);
+
+namespace ForEachLoopNoScopePollution;
+
+use PHPStan\TrinaryLogic;
+use function PHPStan\Testing\assertNativeType;
+use function PHPStan\Testing\assertType;
+use function PHPStan\Testing\assertVariableCertainty;
+
+class ForEachLoopNoScopePollution
+{
+
+	/** @param int $b */
+	public function loopThatIteratesAtLeastOnce(int $a, $b): void
+	{
+		$items = [17 => 'foo', 'bar' => 19];
+
+		foreach ($items as $key => $item) {
+			$a = rand(0, 1);
+			$b = rand(0, 1);
+			$c = rand(0, 1);
+		}
+
+		assertType("17|'bar'", $key);
+		assertNativeType("17|'bar'", $key);
+		assertVariableCertainty(TrinaryLogic::createMaybe(), $key);
+
+		assertType("19|'foo'", $item);
+		assertNativeType("19|'foo'", $item);
+		assertVariableCertainty(TrinaryLogic::createMaybe(), $item);
+
+		assertType('int<0, 1>', $a);
+		assertNativeType('int', $a);
+		assertVariableCertainty(TrinaryLogic::createYes(), $a);
+
+		assertType('int<0, 1>', $b);
+		assertNativeType('int', $b);
+		assertVariableCertainty(TrinaryLogic::createYes(), $b);
+
+		assertType('int<0, 1>', $c);
+		assertNativeType('int', $c);
+		assertVariableCertainty(TrinaryLogic::createYes(), $c);
```

Comment:
This should not be yes but maybe. That's the point of `polluteScopeWithAlwaysIterableForeach: false`.

Please if `for` loop currently behaves like that then please fix the behaviour too.

Reply:
Oh, the docs just talk about variables set in the loop init statements or the key and value variables. If so, my previous PR basically needs to be reverted and the issue it fixed was by design? :)

I can do this in the evening btw.

Reply:
No worries. Should I open a PR for all those tests only? In case they add value, otherwise I don't mind of course.

Reply:
It's fine, I think it's well tested :)